packages/data-point/lib/reducer-types/reducer-helpers/utils/index.test.js

```diff
@@ -0,0 +1,87 @@
+/* eslint-env jest */
+
+const utils = require('./index')
+
+describe('utils#isFalsy', () => {
+  expect(utils.isFalsy(null)).toBe(true)
+  expect(utils.isFalsy(undefined)).toBe(true)
+  expect(utils.isFalsy(NaN)).toBe(true)
+  expect(utils.isFalsy('')).toBe(true)
+  expect(utils.isFalsy('undefined')).toBe(false)
+  expect(utils.isFalsy({})).toBe(false)
+  expect(utils.isFalsy([])).toBe(false)
+  expect(utils.isFalsy(0)).toBe(false)
+  expect(utils.isFalsy(false)).toBe(false)
+  expect(utils.isFalsy(true)).toBe(false)
+})
+
+describe('utils#reducerPredicateIsTruthy', () => {
+  test('with ReducerList', () => {
```

Comment:
This test could be split up to be more descriptive of what exactly is being tested. There are a lot of scenarios and expectations wrapped in a single test, and I think it'd be better if each `expect` was broken into a separate test with a clear description.

For example:

```js
describe('utils#reducerPredicateIsTruthy', () => {
  const reducer = { type: 'ReducerList' }
  test('returns false if reducer is undefined', () => {
    expect(utils.reducerPredicateIsTruthy(reducer, undefined)).toBe(false)
  })
  test('returns false if reducer is false', () => {
    expect(utils.reducerPredicateIsTruthy(reducer, false)).toBe(false)
  })
  test('returns false if reducer is empty string', () => {
    expect(utils.reducerPredicateIsTruthy(reducer, '')).toBe(false)
  })
  // etc
})
```

Reply:
I can add this, but if the message just states the type of the input/ouput, isn't that evident by looking at the test itself?

Reply:
We get a few benefits from splitting it up and writing better descriptions:

- If a single `expect` fails for some reason, it immediately gets narrowed down to the one that failed.
- When `jest` throws the error it'll write the description string in the console, so we'll be able to read the English description of what failed and not read any code to understand the problem.
- If we do a `--verbose` test run we'll get that nice list of descriptions. Some folks do that to read through tests. (Like me 😄 )
- If someone new comes on to the project and isn't very familiar with the `reducerPredicateIsTruthy` method signature, `jest`, or testing in general then these descriptions act as documentation that explains the code.

Yes, it's verbose and duplicative, but it's more explicit and compartmentalized.

Reply:
Thanks, makes a lot of sense!